vault/core.go

```diff
@@ -3967,10 +3967,24 @@ func (c *Core) DetermineRoleFromLoginRequest(mountPoint string, data map[string]
 		Data:       data,
 		Storage:    c.router.MatchingStorageByAPIPath(ctx, mountPoint+"login"),
 	})
-	if err != nil || resp.Data["role"] == nil {
-		return ""
+	if err != nil {
+		return "", err
+	}
+
+	role, ok := resp.Data["role"].(string)
+	if !ok {
+		return "", fmt.Errorf("no role associated with login request")
 	}
-	return resp.Data["role"].(string)
+
+	return role, nil
+}
+
+func (c *Core) RoleBasedQuotasEnabled(ctx context.Context, req *quotas.Request) (bool, error) {
```

Comment:
I think a unit test would be a good idea here.  I'm not confident that this approach is sufficient to find all role-based quota rules.